woodwork/table_accessor.py

```diff
@@ -1081,6 +1082,25 @@ def value_counts(self, ascending=False, top_n=10, dropna=False):
         """
         return _get_value_counts(self._dataframe, ascending, top_n, dropna)
 
+    @_check_table_schema
+    def infer_temporal_frequencies(self, temporal_columns=None):
```

Comment:
Went with "temporal" across this feature instead of "datetime" because it will work for timedelta or datetime. If that's confusing, I'd be willing to switch to just `infer_frequencies` or something else that makes it clear that this is done on the table-level with all qualifying columns.